internal/api/api.go

```diff
@@ -31,6 +31,16 @@ type API struct {
 	db      *storage.Connection
 	config  *conf.GlobalConfiguration
 	version string
+
+	overrideTime func() time.Time
+}
+
+func (a *API) Now() time.Time {
+	if a.overrideTime != nil {
+		return a.overrideTime()
+	}
+
+	return time.Now()
```

Comment:
not too sure why this change is necessary, when will we need to override the time function here?

Reply:
For tests

Reply:
or even set the timebox value to 0 if it's possible in the tests

Reply:
imo the purpose of overriding `time.Now()` is to test the behaviour that the timebox works and it revokes an expired session which is equivalent to setting the timebox to 0 such that any session expires immediately

Reply:
There aren't too many options when testing time. Usually the best way is to provide some way to inject the clock you want to test with... Go is pretty basic so you have to use some `overrideX` style variable to control time. In other languages DI solves this pretty neatly. Other languages also allow you to modify the wall clock, but that's not really an option here as well.

I'm in favor of 0 duration as the 0 value is disallowed from the config (see the new Validate method) as it doesn't convey that much meaning -- is it disabled or not? Milliseconds are also not an option because that can cause flaky tests.

In general all tests involving time should switch to using `api.Now()` instead of `time.Now()` so we can test that behavior properly. Probably a task for the future (though once I'm done with inactivity timeout I'll rewrite all `time.Now()` in this file to use `api.Now()` instead).